ACPs/189-seamless-l1-creation/README.md

```diff
@@ -0,0 +1,129 @@
+
+By introducing a single `CreateL1Tx` transaction, we can simplify the process, reduce overhead, and improve the developer experience for creating L1s.
+
+## **Specification**
+
+### **New Transaction Type**
+
+The following new transaction type is introduced:
+
+```go
+// ChainConfig represents the configuration for a chain to be created
+type ChainConfig struct {
+    // A human readable name for the chain; need not be unique
+    ChainName string `serialize:"true" json:"chainName"`
+    // ID of the VM running on the chain
+    VMID ids.ID `serialize:"true" json:"vmID"`
+    // IDs of the feature extensions running on the chain
+    FxIDs []ids.ID `serialize:"true" json:"fxIDs"`
+    // Byte representation of genesis state of the chain
+    GenesisData []byte `serialize:"true" json:"genesisData"`
+}
+
+// CreateL1Tx is an unsigned transaction to create a new L1 with one or more chains
+type CreateL1Tx struct {
+    // Metadata, inputs and outputs
+    BaseTx `serialize:"true"`
+    
+    // Chain configurations for the L1 (can be multiple)
+    Chains []ChainConfig `serialize:"true" json:"chains"`
+    
+    // Chain where the L1 validator manager lives
+    ManagerChainID ids.ID `serialize:"true" json:"managerChainID"`
+    
+    // Address of the L1 validator manager
+    ManagerAddress types.JSONByteSlice `serialize:"true" json:"managerAddress"`
+    
+    // Initial pay-as-you-go validators for the L1
+    Validators []*L1Validator `serialize:"true" json:"validators"`
+}
+```
+
+The `L1Validator` structure follows the same definition as in ACP-77.
+
+### **Transaction Processing**
+
+When a `CreateL1Tx` transaction is processed, the P-Chain performs the following operations atomically:
+
+1. Create a new L1
+2. Create chain records for each chain configuration in the `Chains` array.  
+3. Set up the L1 validator manager with the specified `ManagerChainID` and `ManagerAddress`  
+4. Register the initial validators specified in the `Validators` array
+
+### IDs
+
+* `subnetID`: The `subnetID` of the L1 is the transaction hash.
+* `blockchainID`: the `blockchainID` for each blockchain is is defined as the SHA256 hash of the 37 bytes resulting from concatenating the 32 byte subnetID with the `0x00` byte and the 4 byte chainIndex (index in the `Chains` array within the transaction)
+* `validationID`: The `validationID` for the initial validators added through CreateL1Tx is defined as the SHA256 hash of the 37 bytes resulting from concatenating the 32 byte subnetID with the `0x01` byte and 4 byte validatorIndex (index in the `Validators` array within the transaction).
+
+### **Restrictions and Validation**
+
+The `CreateL1Tx` transaction has the following restrictions and validation criteria:
+
+1. The `Chains` array must contain at least one chain configuration  
+2. The `ManagerChainID` must be a valid blockchain ID  
+3. The `ManagerAddress` must be a valid address for the specified chain type  
```

Comment:
I don't think this can be enforced on the P-Chain, since it's just a variable length byte array, and the P-Chain does not know of the address types used by other L1s.